io/config.go

```diff
@@ -24,27 +24,53 @@ import (
 	"path/filepath"
 
 	"github.com/apache/brooklyn-client/error_handler"
+	"github.com/apache/brooklyn-client/plugin"
 )
 
 type Config struct {
 	FilePath string
-	Map      map[string]interface{}
+	Model    ConfigModel
+}
+
+type ConfigModel struct {
+	Auth      map[string]Credentials
+	PluginDir string
+	Plugins   map[string]*plugin.PluginMetadata
+	Target    string
+}
+
+type Url struct {
+	Target string
+}
+
+type Credentials struct {
+	Username string
+	Password string
 }
 
 func GetConfig() (config *Config) {
 	// check to see if $BRCLI_HOME/.brooklyn_cli or $HOME/.brooklyn_cli exists
 	// Then parse it to get user credentials
 	config = new(Config)
+	var pluginDir string
 	if os.Getenv("BRCLI_HOME") != "" {
+		pluginDir = filepath.Join(os.Getenv("BRCLI_HOME"), "plugins")
```

Comment:
Would be nice to create $BRCLI_HOME/plugins if it doesn't exist (assuming BRCLI_HOME exists; if not, best complain...)